### PR TITLE
Support for mimic joints and example with gripper.

### PR DESCRIPTION
**Pull request by destogl**
The PR adds functionality for mimic joints and shows its functionality on a parallel gripper example.

﻿Signed-off-by: Denis Štogl <denis@stogl.de>

P.S. After resolving everything here, I will port the same functionality to ign_ros2_control.

### CODE REVIEW

File: gazebo_ros2_control_demos/examples/example_gripper.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Stogl Robotics Consulting UG (haftungsbeschränkt)
```
Comment (ahcorde) [inline gazebo_ros2_control_demos/examples/example_gripper.cpp:1]: do you mind to add yourself as an autor in the file but use the open robotics copyright ?

File: gazebo_ros2_control_demos/launch/gripper_mimic_joint_example.launch.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# Copyright 2022 Stogl Robotics Consulting UG (haftungsbeschränkt)
```
Comment (ahcorde) [inline gazebo_ros2_control_demos/launch/gripper_mimic_joint_example.launch.py:1]: same here

**Review by ahcorde — CHANGES_REQUESTED** [review]
Thank you for the contribution!

can you review linters?

File: gazebo_ros2_control_demos/examples/example_gripper.cpp (hunk 1 above)
Comment (ahcorde) [reply gazebo_ros2_control_demos/examples/example_gripper.cpp:1]: Sorry I was applying the Ignition review guidelines, it's actually not an issue here. In Ignition we ask that people voluntarily assign their copyright to OSRF which allows the Foundation more flexibility in the future.

Anyhow @tfoote or @chapulina can add here more details

Comment (destogl) [reply gazebo_ros2_control_demos/examples/example_gripper.cpp:1]: OK, I'll do this for Ignition plugin

**Comment by destogl** [comment]
@ahcorde this is ready to be merged. There is an issue currently with testing repository for ros2 where ros2_controller are missing.

File: gazebo_ros2_control_demos/launch/gripper_mimic_joint_example.launch.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    gazebo = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            [FindPackageShare("gazebo_ros"), "/launch", "/gazebo.launch.py"]
+        ),
+    )
+
+    spawn_entity = Node(package='gazebo_ros', executable='spawn_entity.py',
+                        arguments=['-topic', 'robot_description',
+                                   '-entity', 'cartpole'],
```
Comment (ahcorde) [inline gazebo_ros2_control_demos/launch/gripper_mimic_joint_example.launch.py:57]:
```suggestion
                                   '-entity', 'gripper'],
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
Just a minor comment in the launch file

**Comment by destogl** [comment]
@ahcorde just updated!

**Review by ahcorde — APPROVED** [review]

**Comment by destogl** [comment]
you will have to restart CI, something went bad there during environment setup.

**Comment by ahcorde** [comment]
Thank you @destogl !

Looking forward to see this feature in `ign_ros2_control`. Let me know if you don't have time and I will do it.

**Pull request merged**